ex03.exs

```diff
@@ -45,17 +45,37 @@ defmodule Ex03 do
   six helper functions.
 
   35 points:
-     it works and passes all tests:    25
-     it contains no conditional logic:  3
-     it is nicely structured            7
+	 it works and passes all tests:    25
+	 it contains no conditional logic:  3
+	 it is nicely structured            7
   """
 
-  def pmap(collection, process_count, function) do
-    # your code goes here
-    # I'm hoping to see a simple pipeline as the body of this function...
-  end
-
-  # and here...
+	def pmap(collection, process_count, function) do
+		splitCollection(collection, process_count)
+		|> mapCollections(function)
+	end
+	
+	#Divides the collection into n groups where n is process_count
+	def splitCollection(collection, process_count) do
+		collection |> Enum.chunk_every(trunc(Enum.count(collection) / process_count))
```

Comment:
There's a bug here:

     Ex03.splitCollection(1..10, 4) |> Enum.count

it needs to be `trunc((Enum.count(collection) + process_count - 1) / process_count)`

# -2